src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Model/GenerationSpec.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using DotnetRuntime.SourceGenerators;
+
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
+{
+    [DebuggerDisplay("Type index count={TypeList.Count}")]
+    internal sealed record GenerationSpec
+    {
+        public required ImmutableEquatableArray<string> TypeNamespaces { get; init; }
+        public required ImmutableEquatableArray<TypeSpec> TypeList { get; init; }
+
+        public MethodsToGen_CoreBindingHelper MethodsToGen_CoreBindingHelper { get; init; }
+        public MethodsToGen_ConfigurationBinder MethodsToGen_ConfigurationBinder { get; init; }
+        public MethodsToGen_Extensions_OptionsBuilder MethodsToGen_OptionsBuilderExt { get; init; }
+        public MethodsToGen_Extensions_ServiceCollection MethodsToGen_ServiceCollectionExt { get; init; }
+
+        public required ImmutableEquatableArray<TypeWithChildrenSpec> TypesForGen_ConfigurationBinder_Bind_instance { get; init; }
+        public required ImmutableEquatableArray<TypeWithChildrenSpec> TypesForGen_ConfigurationBinder_Bind_instance_BinderOptions { get; init; }
+        public required ImmutableEquatableArray<TypeWithChildrenSpec> TypesForGen_ConfigurationBinder_Bind_key_instance { get; init; }
+
+        public required bool GraphContainsEnum { get; init; }
+        public required ImmutableEquatableArray<TypeSpec> TypesForGen_CoreBindingHelper_BindCoreUntyped { get; init; }
+        public required ImmutableEquatableArray<TypeSpec> TypesForGen_CoreBindingHelper_GetCore { get; init; }
+        public required ImmutableEquatableArray<TypeSpec> TypesForGen_CoreBindingHelper_GetValueCore { get; init; }
+        public required ImmutableEquatableArray<TypeWithChildrenSpec> TypesForGen_CoreBindingHelper_BindCore { get; init; }
+        public required ImmutableEquatableArray<ObjectSpec> TypesForGen_CoreBindingHelper_Initialize { get; init; }
+        public required ImmutableEquatableArray<ParsableFromStringSpec> TypesForGen_CoreBindingHelper_ParsePrimitive { get; init; }
+
+        // TODO: add ImmutableEquatableDictionary to be supplied by the parser.
```

Comment:
Perhaps the current approach of delaying creation of the dictionary is better. That way you avoid creating one if incremental values match and you don't need to retrigger the emitter.